xds/src/main/java/io/grpc/xds/LookasideLb.java

```diff
@@ -49,169 +48,134 @@
 import io.grpc.xds.LocalityStore.LocalityStoreFactory;
 import io.grpc.xds.XdsClient.EndpointUpdate;
 import io.grpc.xds.XdsClient.EndpointWatcher;
+import io.grpc.xds.XdsClient.RefCountedXdsClientObjectPool;
+import io.grpc.xds.XdsClient.XdsClientFactory;
 import io.grpc.xds.XdsLoadBalancerProvider.XdsConfig;
+import io.grpc.xds.XdsSubchannelPickers.ErrorPicker;
 import java.util.Collections;
 import java.util.List;
 import java.util.Map;
 import java.util.logging.Logger;
 import javax.annotation.Nullable;
 
-/** Lookaside load balancer that handles balancer name changes. */
-final class LookasideLb extends ForwardingLoadBalancer {
+/** Lookaside load balancer that handles EDS config. */
+final class LookasideLb extends LoadBalancer {
```

Comment:
Isn't calling it `EdsLoadBalancer` more appropriate now? Since it no longer do look aside by itself (and implementation should not have knowledge of looking aside, it's just adding watchers to `XdsClient` to receive `EndpointUpdate`). I understand this class will not be instantiated directly by`AutoConfiguredLoadBalancerProvide` the same way as `CdsLoadBalancer` does, but I think calling it `EdsLoadBalancer` is still ok and better than now.

Reply:
I don’t care too much what the name should be but this is very much a “look-aside” load balancer - in fact the last and most important component. Look-aside as opposed to a proxy lb . Since it gets the endpoints and picks one it *is* a look-aside lb even if looking aside is done elsewhere.

Reply:
I don’t care too much what the name should be but this is very much a “look-aside” load balancer - in fact the last and most important component. Look-aside as opposed to a proxy lb . Since it gets the endpoints and picks one it *is* a look-aside lb even if looking aside is done elsewhere.

Reply:
We already have `XdsLoadBalancer` which is an alias to `EdsLoadBalancer`. `LookasideLb` is name as a component of `XdsLoadBalancer` in contrast with the other component `fallbackLb`. Both names have pros and cons, I tend to keep it unchanged unless it's causing bigger confusion.